src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1054,10 +1054,11 @@ def enable_model_cpu_offload(self, gpu_id: Optional[int] = None, device: Union[t
         device = torch.device(f"{device_type}:{self._offload_gpu_id}")
         self._offload_device = device
 
-        self.to("cpu", silence_dtype_warnings=True)
-        device_mod = getattr(torch, device.type, None)
-        if hasattr(device_mod, "empty_cache") and device_mod.is_available():
-            device_mod.empty_cache()  # otherwise we don't see the memory savings (but they probably exist)
+        if self.device.type != "cpu":
```

Comment:
I don't think this check is needed because we need to move all the components to cpu regardless